### PR TITLE
Draft implementation of correct UTF-8 parsing in textUtf8, see issue #509

### PR DESCRIPTION
**Pull request by SLiV9**
~~This is a draft and should probably not be merged because it breaks backwards compatibility.~~ ~~(Also I've added example carts to _examples/utf8_ for testing purposes, that can be removed.)~~

This updates the `textUtf8` function in _runtime.ts_ to correctly parse its input as UTF-8 and send an array of codepoints to `drawText` in _framebuffer.ts_, instead of creating mojibake. It also updates `drawText` to draw tofu for Unicode codepoints that are not supported and a replacement character (U+FFFD) for invalid Unicode codepoints. See https://github.com/aduros/wasm4/issues/509#issuecomment-1224263429 for details.

Still to do:
* [x]  Implement backwards compatibility.
* [x]  Implement this fix in the native runtime.
* [x]  Update _site/docs/guides/text.md_ with code snippets for different langauges.

### CODE REVIEW

File: runtimes/web/src/runtime.ts [hunk 1]
```diff
@@ -194,9 +194,11 @@ export class Runtime {
         this.framebuffer.drawText(text, x, y);
     }
 
-    textUtf8 (textPtr: number, byteLength: number, x: number, y: number) {
-        const text = new Uint8Array(this.memory.buffer, textPtr, byteLength);
-        this.framebuffer.drawText(text, x, y);
+    textUtf8 (strUtf8Ptr: number, byteLength: number, x: number, y: number) {
+        const strUtf8 = new Uint8Array(this.memory.buffer, strUtf8Ptr, byteLength);
+        const str = new TextDecoder().decode(strUtf8);
+        let codepoints: number[] = [...str].map(x => x.codePointAt(0));
```
Comment (MaxGraey) [inline runtimes/web/src/runtime.ts:200]:
```suggestion
        let codepoints = Array.from(str);
```

**Comment by aduros** [comment]
Thanks Sliv, I'll take a closer look in the next few days.

What would the impact be if we didn't want to keep the old behavior for backwards compat? Would it only affect existing carts that use textUtf8/16 to draw the extended ASCII glyphs?

**Comment by SLiV9** [comment]
It would affect all existing carts written in Zig, Rust, Go and Odin that drew any characters above 0x7f. Those characters become question marks.

**Comment by aduros** [comment]
Closing this as superceded by #591. The naming of `textUtf8/16` isn't ideal, but I think it's something we can live with.

**Pull request closed without merging**